File: AudiobookManager/AudiobookManager.Scraping/Models/MetadataMultiSourceSearchResult.cs
namespace AudiobookManager.Scraping.Models;

public class MetadataMultiSourceSearchResult
{
    public IList<MetadataSearchResult> Results { get; set; } = new List<MetadataSearchResult>();
    public IList<MetadataSourceSearchStatus> SourceStatuses { get; set; } = new List<MetadataSourceSearchStatus>();
}

public class MetadataSourceSearchStatus
{
    public string Source { get; set; } = "";
    public bool Success { get; set; }
    public int ResultCount { get; set; }
    public string? Error { get; set; }
}

File: client/src/types/MetadataMultiSourceSearchResult.ts
import type { MetadataSearchResult } from "./MetadataSearchResult";

// AudiobookManager.Scraping/Models/MetadataMultiSourceSearchResult.cs: both fields default to a
// non-null empty list and are never set to null.
export interface MetadataSourceSearchStatus {
  source: string;
  success: boolean;
  resultCount: number;
  error?: string;
}

export interface MetadataMultiSourceSearchResult {
  results: MetadataSearchResult[];
  sourceStatuses: MetadataSourceSearchStatus[];
}

export type { MetadataMultiSourceSearchResult as default };
